### ISSUE TITLE
Add 'intaked' search filter

### ISSUE THREAD

**Issue by samschlicht** [talentbeyondboundaries/tbbtalentv2#304]
As a TC admin
I would like to be able to search on the basis of intaked status
So that I can efficiently make lists, e.g., choosing only intaked candidates for an urgent opportunity, or finding highly qualified candidates that have not been intaked yet, for team attention

**Comment by samschlicht** [comment]
Based on consultation with Mariam and Zeynep and my own thinking, this should be a dropdown filter matching the options in the 'Intaked' column:
—
Mini
Full

**Comment by samschlicht** [comment]
Just noting that in my view this issue is blocked on #493 — it can be done, but I'm not sure we should be coding more features onto a transient field. Hope we can take the broader intake issue into consideration in a not-too-distant release. The same could also be said for #160 and #486.

**Comment by cazcam34** [comment]
Need #493 merged into staging so I can merge into this branch, then I can add new flyway with new fields and test and then it should be ready

**Comment by cazcam34** [comment]
We won't need 'input external input' button for Fragomen as they'll be inputting at the time of interview so can use the regular complete button. Create new issue for YAML for that external input button?

**Comment by cazcam34** [comment]
Also replaced old intaked column to use new fields instead of transient field. Problem with this column is it isn't sortable, as we are showing a combined view of what intakes are completed 'Mini, Full' or just 'Mini' or 'Full'. As I have to create the string to display, it will always have Mini at the start even if it's Mini & Full.
Also this combined field isn't on the database anymore and is just put together based on the fields on the candidate object on the front end. So I can't sort this on the backend.

**Comment by cazcam34** [comment]
Added a tooltip with dates as per #486

**Comment by cazcam34** [comment]
Get into test environment and get back to Mariam/Zeynep telling them its in our next release + gather any final feedabck.

**Comment by cazcam34** [comment]
Video: https://app.screencastify.com/v3/watch/7oAheV8qgbgrbMlhQTvO

**Comment by cazcam34** [comment]
Hi mariam and Zeynep (and anyone else interested!), we have a new feature you've requested and we'd love to give you a sneak peak and capture any feedback. It's currently in our test environment for you to properly give a go and if it's all is good, it will be in our next release later this month. 
Feature: Search candidates by intake status, so whether they have their mini and/or full intake completed or not. Once searched you can view these statuses using the intake status column and if hovered over you will be able to see the dates of completion for the full & mini intakes. Along with this search feature, we slightly updated the appearance of the intake forms to feature just one button - **Complete** or **Update**. An intake will only be able to be completed once, and clicking this button will store the data that is needed for the search filter as mentioned above. Once an intake has been completed, the data is stored and the button then becomes an Update button. Clicking the Update button will just create a candidate note with the audit data of that click (who/when), and then more details can be added to that note keep track of what was updated. I made a short video running through these changes, and as mentioned they are currently in the test environment ready for testing so please have a look! Let me know if you have any questions/feedback :)
 
LINK TO VIDEO

A reminder how to get into the test environment, go to tctalent-test.org/admin-portal

**Comment by cazcam34** [comment]
Feedback:
- Just show latest intake in column, BUT highlight if full but no mini so they can see if they need to check it.
- After above, make a latest intake data column based on the latest intake. Does that give Mariam what she wants in terms of a filter.
- Check how we got the names of people with intakes, that we intaked. Are there any others in drive?
- If we discover intakes that aren't in TC we can add button back.
- 

UPDATE!
- We haven't ran the new SQL update to populate the intake columns yet, so we can't check prod for intakes not completed as none of them have that field completed.

**Comment by cazcam34** [comment]
SQL work around as we don't yet have the intake audit fields in prod, but those are populated using the notes so we can put together a query using that: 
`select candidate_number, folderlink from candidate c where id not in (
    select candidate_id from candidate_note n where (lower(n.title) LIKE '%full intake interview completed%'
        OR lower(n.title) LIKE '%full intake took place%'
        OR lower(n.title) LIKE '%mini intake interview completed%'
        OR lower(n.title) LIKE '%mini intake took place%' ))
                            and status = 'active'
                            and folderlink is not null;`
TOTAL: 1,514 have NO intake note (so no populated intake field(s)) but are active and have a folder link.

Note: 17,901 have an intake note that will be used to populate the intake fields.

Updated to only show ones with job opps:
`select candidate_number, folderlink from candidate c where id not in (
    select candidate_id from candidate_note n where (lower(n.title) LIKE '%full intake interview completed%'
        OR lower(n.title) LIKE '%full intake took place%'
        OR lower(n.title) LIKE '%mini intake interview completed%'
        OR lower(n.title) LIKE '%mini intake took place%' ))
                            and status = 'active'
                            and folderlink is not null
and (select count(*) from candidate_opportunity where candidate_id = c.id) > 0;`
Total = 269

**Comment by cazcam34** [comment]
Hi Mariam and Zeynep! Thank you so much for your feedback, we really appreciate you going into the test environment and giving it a good testing! 
Notes on your feedback:
- We definitely can update the column to display the LATEST intake only, as you mention there shouldn't be a mini intake without a full prior. However we thought to add a little 'note' on the column if there is a full without a mini, eg "Full (no mini)", so that the user can check if the complete button was missed at the mini intake stage. 
- To help with your filtering on intake date, we thought we could add a LATEST intake completed date column. This can then be sortable, so you can do your search and then sort by date. Hopefully this will meet your filtering needs for intake date?
Regarding the interns coming on:
- Chatting with John about when we previously did the inputting of external intakes into the TC (3 or so years ago?!) we are fairly sure we just sorted by active candidates, and only inputted their intakes into the TC. xxxxxxx
- Yes we removed the input external intake button, as we want people to be completing intakes on the TC directly. However if they are going to inputting more old intakes, we can definitely just add the button back in.

**Comment by cazcam34** [comment]
UPDATE: feedback implemented in new PR which is in now.
Along with the feedback came another query from Mariam regarding interns inputting old intakes. 
- Provided her with some CSV files using the SQL above. Also requested non active statuses.
- Regarding the timeline, we're aiming to have the interns complete the intake transfer by **March 28**. If it's possible to keep the "input external intake" available until then, it would greatly assist us.
- Access issue for the interns to access the google drives (we will have interns inputting old google drive intake docs into the TC, I provided her with a list of candidates and their google drive links but of course they won't be able to access those links from external accounts) Sam assisting with this.

**Comment by cazcam34** [comment]
Get in touch Mariam for more details on the Phase 2 - what do these emails include, are they standardised or personalised? Can we do a single device shared to send them?

**Comment by sadatmalik** [comment]
Checked with Mariam - the external intake has not been completed by the 28th as initially expected. Therefore we need to keep the "Input External Intake" in the 2.2.1 release.
